### PR TITLE
Adding documentation and examples for Transparent Data Encryption, En…

### PR DESCRIPTION
**Pull request by mihymel**
…cryption Protector, Server Key, and Servers

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: arm-sql/2015-05-01-preview/swagger/serverKeys.json [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+        "kind": {
+          "description": "Kind of encryption protector. This is metadata used for the Azure portal experience.",
+          "enum": [
+            "None",
+            "azurekeyvault",
+            "servicemanaged"
+          ],
+          "type": "string"
+        },
+        "location": {
```
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/serverKeys.json]: Just to confirm, by adding location  you dont intend to make this a tracked resource.

File: arm-sql/2015-05-01-preview/swagger/sql.core.json [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+            "$ref": "../examples/ServerGet.json"
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "Servers"
+        ],
+        "description": "Creates or updates a server.",
+        "operationId": "Servers_CreateOrUpdate",
```
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/sql.core.json]: If you already have an update operation you probably don't need a createOrUpdate

File: arm-sql/2015-05-01-preview/swagger/sql.core.json [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+          },
+          "202": {
+            "description": "Accepted"
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
```
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/sql.core.json]: Please provide ```x-ms-examples``` here.

File: arm-sql/2015-05-01-preview/swagger/sql.core.json [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+          "$ref": "#/definitions/ResourceIdentity",
+          "description": "The Azure Active Directory identity of the server.",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ]
+        },
+        "kind": {
+          "description": "Kind of sql server. This is metadata used for the Azure portal experience.",
+          "enum": [
```
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/sql.core.json]: please provide an ```x-ms-enums``` section with appropriate options here,

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks in pretty good shape. Left a few comments.

File: arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+              "modelAsString": true
+            }
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "The requested encryption protector resource state.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EncryptionProtector"
```
Comment (mihymel) [inline arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json]: I double checked to be safe - ServerKeyType and ServerKeyName are the only properties of EncryptionProtector that are not readonly. This is what we want.

File: arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/encryptionProtector": {
+      "get": {
+        "tags": [
+          "EncryptionProtectors"
+        ],
+        "description": "Returns a list of the server encryption protectors",
+        "operationId": "Servers_ListEncryptionProtectors",
```
Comment (mihymel) [inline arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json]: EncryptionProtector is a singleton, so this is following the pattern for naming singleton operations. See [blob auditing policies](https://github.com/Azure/azure-rest-api-specs/blob/master/arm-sql/2015-05-01-preview/swagger/blobAuditingPolicies.json) for example of the pattern.

File: arm-sql/2015-05-01-preview/swagger/serverKeys.json (hunk 1 above)
Comment (mihymel) [inline arm-sql/2015-05-01-preview/swagger/serverKeys.json]: Correct. This is not a tracked resource.

File: arm-sql/2015-05-01-preview/swagger/sql.core.json (hunk 2 above)
Comment (mihymel) [inline arm-sql/2015-05-01-preview/swagger/sql.core.json]: "CreateOrUpdate" is for PUT, "Update" is for PATCH.

File: arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json [hunk 7]
```diff
@@ -0,0 +1,355 @@
...
+      },
+      "put": {
+        "tags": [
+          "EncryptionProtectors"
+        ],
+        "description": "Updates an existing encryption protector.",
+        "operationId": "Servers_CreateOrUpdateEncryptionProtector",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
```
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json]: Maintain consistent parameter ordering, either place "ResourceGroupParameter" and "ApiVersionParameter" together at the beginning or at the end of the parameters section.

File: arm-sql/2015-05-01-preview/swagger/sql.core.json (hunk 2 above)
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/sql.core.json]: Let me rephrase that, do you think you will need a ```CreateOrUpdate``` even if you had ```Create``` and ```Update``` operations?
```CreateOrUpdate``` is useful if the update operation only allows an update on the ```tags``` property. If you wish to expose a PATCH operation that allows updating of other properties, its better to have a separate ```Update``` operation like you have here.

File: arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json (hunk 7 above)
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json]: Why is this? You previously told us that SubscriptionId and ApiVersion should be together since they are client params, not method params. Did this change?

File: arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json [hunk 8]
```diff
@@ -0,0 +1,355 @@
...
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
```
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json]: It's in sql.core.json

File: arm-sql/2015-05-01-preview/swagger/sql.core.json (hunk 2 above)
Comment (jaredmoo) [inline arm-sql/2015-05-01-preview/swagger/sql.core.json]: There is no Create operation. Just Servers_CreateOrUpdate and Servers_Update

File: arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json (hunk 7 above)
Comment (dsgouda) [inline arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json]: Whoops I confused ```SubscriptionId`` with ```ResourceGroupName```, sorry about that, please disregard this

**Comment by jaredmoo** [comment]
Design note: arguably `az sql db list-capabilities --edition Standard` should instead be `az sql db show-capabilities --edition Standard`. However I don't think the distinction is worth implementing an additional command.

**Comment by jaredmoo** [comment]
@nathannfan to review

File: arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json [hunk 9]
```diff
@@ -0,0 +1,359 @@
...
+            "in": "path",
+            "description": "The name of the encryption protector to be updated.",
+            "required": true,
+            "type": "string",
+            "enum": [
+              "current"
+            ],
+            "x-ms-enum": {
+              "name": "EncryptionProtectorName",
+              "modelAsString": true
```
Comment (nathannfan) [inline arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json]: I assume this might change?

**Review by nathannfan — APPROVED** [review]

File: arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json (hunk 9 above)
Comment (mihymel) [inline arm-sql/2015-05-01-preview/swagger/encryptionProtectors.json]: EncryptionProtector is a singleton now and we intend for it to be that way for the unforeseeable future. Technically (with some changes) the API is designed to support multiple EncryptionProtectors, so I used modelAsString.

**Comment by mihymel** [comment]
I removed "x-ms-enum" for some parameters where an acceptable enum value included an empty string because in this case this resulted in errors in the generated .NET SDK. The generated .NET SDK generates an enum class where the enum with value "" doesn't have a name: e.g.  public const string  = "";

**Comment by nathannfan** [comment]
@dsgouda the validation error for syntax isn't very clear.  Is this a legitimate syntax error in the swagger?

**Comment by dsgouda** [comment]
If there are no explicit properties you wish to add under "x-ms-enums", you can remove it or just put an empty object there, I leave it to your discretion.

**Comment by dsgouda** [comment]
@nathannfan which specific error is this related too, I can try and help you out with it.

**Comment by nathannfan** [comment]
@dsgouda thanks, I'm referring to https://travis-ci.org/Azure/azure-rest-api-specs/jobs/236527345#L1056

**Review by jaredmoo — APPROVED** [review]

**Review by dsgouda — APPROVED** [review]
